packages/valory/skills/decision_maker_abci/tests/test_redeem_info.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Tests for redeem info of decision_maker skill."""
+
+import pytest
+from hexbytes import HexBytes
+
+from packages.valory.skills.decision_maker_abci.redeem_info import (
+    Condition,
+    FPMM,
+    Question,
+    Trade,
+)
+
+
+class TestCondition:
+    """Test condition."""
+
+    def test_initialization(self) -> None:
+        """Test initialization."""
+        condition = Condition(
+            id="0x00000000000000001234567890abcdef", outcomeSlotCount=2
+        )
+        condition.__post_init__()
+
+        assert condition.outcomeSlotCount == 2
+        assert condition.id == HexBytes("0x00000000000000001234567890abcdef")
+        assert condition.index_sets == [1, 2]
+
+
+class TestQuestion:
+    """Test Question."""
+
+    @pytest.mark.parametrize(
+        "name, id",
+        [
+            [
+                "id as bytes",
+                b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+            ],
+            ["id as string", "0x00000000000000001234567890abcdef"],
+        ],
+    )
+    def test_initialization(self, name: str, id: bytes) -> None:
+        """Test initialization."""
+        question = Question(id=id, data="dummy_data")
+
+        question.__post_init__()
```

Comment:
The post init is automatically called. I would suggest that you check whether its result is the expected one, i.e., if the given `id` is a string, check that it is bytes after initialization.